File: vben-admin/apps/web-antd/src/api/ydsj.ts
import { requestClient } from '#/api/request';

export interface YDSJConfig {
  base_url: string;
  token: string;
  uid: string;
  key: string;
  price_multiple: number;
  xbd_morning_price: number;
  xbd_exercise_price: number;
  real_cost_multiple: number;
}

export interface YDSJOrder {
  id: number;
  yid: string;
  uid: number;
  school: string;
  user: string;
  pass: string;
  distance: string;
  is_run: number;
  run_type: number;
  start_hour: string;
  start_minute: string;
  end_hour: string;
  end_minute: string;
  run_week: string;
  status: number;
  remarks: string;
  info: string;
  tmp_info: string;
  fees: string;
  real_fees: string;
  refund_money: string;
  addtime: string;
}

// 配置
export async function ydsjConfigGetApi() {
  return requestClient.get<YDSJConfig>('/ydsj/config');
}
export async function ydsjConfigSaveApi(data: YDSJConfig) {
  return requestClient.post('/ydsj/config', data);
}

// 价格
export async function ydsjGetPriceApi(run_type: number, distance: number) {
  return requestClient.post<{ price: number }>('/ydsj/price', { run_type, distance });
}

// 订单列表
export async function ydsjOrderListApi(params: {
  page: number;
  limit: number;
  searchType?: string;
  keyword?: string;
  status?: string;
}) {
  return requestClient.get<{ list: YDSJOrder[]; total: number }>('/ydsj/orders', { params });
}

// 下单
export async function ydsjAddOrderApi(form: Record<string, any>) {
  return requestClient.post('/ydsj/add', { form });
}

// 退款
export async function ydsjRefundOrderApi(id: number) {
  return requestClient.post('/ydsj/refund', { id });
}

// 切换跑步状态
export async function ydsjToggleRunApi(id: number) {
  return requestClient.post('/ydsj/toggle-run', { id });
}

File: go-api/internal/modules/ydsj/ydsj.go
package ydsj

import (
	_ "embed"
	"net/http"
	"time"
)

//go:embed ydsj_schools.json
var ydsjSchoolsJSON []byte

type YDSJConfig struct {
	BaseURL          string  `json:"base_url"`           // 上游API地址 如 http://103.236.75.71:7799/LearnExp
	UID              string  `json:"uid"`                // 对接站用户UID
	Key              string  `json:"key"`                // 对接站密钥
	Token            string  `json:"token"`              // (旧字段，保留兼容)
	PriceMultiple    float64 `json:"price_multiple"`     // 运动世界价格倍率
	XbdMorningPrice  float64 `json:"xbd_morning_price"`  // 小步点晨跑价格
	XbdExercisePrice float64 `json:"xbd_exercise_price"` // 小步点课外跑价格
	RealCostMultiple float64 `json:"real_cost_multiple"` // 实际扣费倍率
}

type YDSJOrder struct {
	ID          int    `json:"id"`
	YID         string `json:"yid"`
	UID         int    `json:"uid"`
	School      string `json:"school"`
	User        string `json:"user"`
	Pass        string `json:"pass"`
	Distance    string `json:"distance"`
	IsRun       int    `json:"is_run"`
	RunType     int    `json:"run_type"`
	StartHour   string `json:"start_hour"`
	StartMinute string `json:"start_minute"`
	EndHour     string `json:"end_hour"`
	EndMinute   string `json:"end_minute"`
	RunWeek     string `json:"run_week"`
	Status      int    `json:"status"`
	Remarks     string `json:"remarks"`
	Info        string `json:"info"`
	TmpInfo     string `json:"tmp_info"`
	Fees        string `json:"fees"`
	RealFees    string `json:"real_fees"`
	RefundMoney string `json:"refund_money"`
	Addtime     string `json:"addtime"`
}

type YDSJService struct {
	client *http.Client
}

var ydsjService = &YDSJService{client: &http.Client{Timeout: 30 * time.Second}}

func YDSJ() *YDSJService {
	return ydsjService
}
